packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain/useMaxAmount.ts

```diff
@@ -105,9 +105,26 @@ export function useMaxAmount() {
     if (!isDepositMode) {
       return undefined
     }
+    if (typeof estimatedChildChainGasFees === 'undefined') {
+      return undefined
+    }
+    if (typeof nativeCurrencyMaxAmount === 'undefined') {
+      return undefined
+    }
+
+    if (nativeCurrency.isCustom) {
+      return String(
+        Number(nativeCurrencyMaxAmount) - estimatedChildChainGasFees * 1.4
```

Comment:
I think this could fail, if `nativeCurrencyMaxAmount` is a huge number, then casting it to number would overflow

Reply:
This is how we've been handling numbers here before, it's a formatted number so it should be fine for majority of the cases, but we could have a ticket to refactor this to return bignumber?